solus.go

```diff
@@ -106,7 +107,13 @@ func WithLogger(logger Logger) ClientOption {
 }
 
 // NewClient create and initialize Client instance.
-func NewClient(baseURL *url.URL, a Authenticator, opts ...ClientOption) (*Client, error) {
+func NewClient(
+	baseURL *url.URL,
+	a Authenticator,
+	retries int,
```

Comment:
Please use `ClientOption` instead of adding two new arguments.

Reply:
added `SetRetryPolicy` option to set both `Retry` and `RetryAfter`